src/GrpcCredentialsHelper.php

```diff
@@ -42,53 +42,82 @@
      *
-     *     @var \Google\Auth\CredentialsLoader $credentialsLoader
-     *          A user-created CredentialsLoader object. Defaults to using
-     *          ApplicationDefaultCredentials
-     *     @var boolean $enableCaching
-     *          Enable caching of access tokens. Defaults to true
+     *     @type string $serviceAddress
+     *           Required. The domain name of the API remote host.
+     *     @type mixed $port
+     *           Required. The port on which to connect to the remote host.
+     *     @type string[] $scopes
+     *           Optional. A list of scopes required for API access.
+     *           Exactly one of $scopes or $credentialsLoader must be provided.
+     *           NOTE: if $credentialsLoader is provided, this argument is ignored.
+     *     @type \Google\Auth\CredentialsLoader $credentialsLoader
+     *           Optional. A user-created CredentialsLoader object. Defaults to using
+     *           ApplicationDefaultCredentials with the provided $scopes argument.
+     *           Exactly one of $scopes or $credentialsLoader must be provided.
+     *     @type \Grpc\Channel $channel
+     *           Optional. A `Channel` object to be used by gRPC. If not specified, a channel will be constructed.
+     *     @type \Grpc\ChannelCredentials $sslCreds
+     *           Optional. A `ChannelCredentials` object for use with an SSL-enabled channel.
+     *           Default: a credentials object returned from
+     *           \Grpc\ChannelCredentials::createSsl()
+     *           NOTE: if the $channel optional argument is specified, then this option is unused.
+     *     @type bool $forceNewChannel
+     *           Optional. If true, this forces gRPC to create a new channel instead of using a persistent channel.
+     *           Defaults to false.
+     *           NOTE: if the $channel optional argument is specified, then this option is unused.
+     *     @type boolean $enableCaching
+     *           Optional. Enable caching of access tokens. Defaults to true.
      * }
-     *
      */
-    public function __construct($scopes, $optionalArgs = [])
+    public function __construct($args)
     {
+        $this->validateNotNull($args, [
+            'serviceAddress',
+            'port',
+        ]);
+
         $defaultOptions = [
-            'credentialsLoader' => null,
+            'forceNewChannel' => false,
             'enableCaching' => true,
         ];
-        $opts = array_merge($defaultOptions, $optionalArgs);
-        $cachingOptions = array_diff_key($opts, $defaultOptions);
-
-        $this->scopes = $scopes;
+        $args = array_merge($defaultOptions, $args);
 
-        if (isset($opts['credentialsLoader'])) {
-            $credentialsLoader = $opts['credentialsLoader'];
-        } else {
-            $credentialsLoader = $this->getADCCredentials($scopes);
+        if (empty($args['credentialsLoader'])) {
```

Comment:
In this case, `scopes` is ignored (see https://github.com/googleapis/gax-php/pull/91/files/225501ad7904fa26b397e7e4132e730c93239c7e#diff-223c9981f8726481e32c3d2c29471adeR64)